ObjectiveCloudant/Operations/CDTQueryViewOperation.m

```diff
@@ -0,0 +1,295 @@
+    } else if (statusCode / 100 == 2) {
+        // process the things!
+        NSError *jsonError;
+        NSDictionary<NSString *, NSObject *> *responseJson =
+            [NSJSONSerialization JSONObjectWithData:responseData options:0 error:&jsonError];
+
+        if (!responseJson) {
+            [self callCompletionHandlerWithError:jsonError];
+        } else {
+            if (self.viewRowBlock) {
+                for (NSDictionary<NSString *, NSObject *> *row in(
+                         NSArray<NSDictionary<NSString *, NSObject *> *> *)responseJson[@"rows"]) {
+                    self.viewRowBlock(row);
+                }
+            }
+            if (self.queryViewCompletionBlock) {
+                self.queryViewCompletionBlock(nil);
+            }
+        }
+
+    } else {
+        NSString *json = [[NSString alloc] initWithData:responseData encoding:NSUTF8StringEncoding];
+        NSString *msg = [NSString
+            stringWithFormat:@"Querying view failed with %ld %@.", (long)statusCode, json];
+        NSDictionary *userInfo = @{NSLocalizedDescriptionKey : NSLocalizedString(msg, nil)};
+        error = [NSError errorWithDomain:CDTObjectiveCloudantErrorDomain
+                                    code:CDTObjectiveCloudantErrorQueryViewFailed
+                                userInfo:userInfo];
+
+        [self callCompletionHandlerWithError:error];
+    }
+}
+
++ (BOOL)isValidStringOrJsonArray:(NSObject *)obj
+{
+    if ([obj isKindOfClass:[NSArray class]]) {
+        if (![NSJSONSerialization isValidJSONObject:obj]) {
+            return NO;
+        }
+    } else if (![obj isKindOfClass:[NSString class]]) {
+        return NO;
+    }
+
+    return YES;
+}
+
++ (NSString *)stringForBoolean:(BOOL)boolean
+{
+    if (boolean) {
+        return @"true";
+    } else {
+        return @"false";
+    }
+}
+
++ (NSString *)stringForObject:(NSObject *)obj
+{
+    if ([obj isKindOfClass:[NSArray class]]) {
+        NSError *error;
+        NSData *data = [NSJSONSerialization dataWithJSONObject:obj options:0 error:&error];
+
+        if (error) {
+            // well we need to error here
+            @throw [NSException exceptionWithName:@"JSONSerialisationError"
+                                           reason:@"Could not seralise array into json"
+                                         userInfo:nil];
+        }
+
+        return [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
+
+    } else if ([obj isKindOfClass:[NSString class]]) {
+        return [NSString stringWithFormat:@"\"%@\"", (NSString *)obj];
+    } else {
+        return [NSString stringWithFormat:@"\"%@\"", [obj description]];
+    }
+}
+
++ (NSURLQueryItem *)QueryItemForStaleState:(CDTStale)stale
+{
+    NSString *staleStr;
+
+    switch (stale) {
+        case CDTStaleViewNo:
+            return nil;
+            break;
+        case CDTStaleViewOk:
+            staleStr = @"ok";
+            break;
+
```

Comment:
Is this simpler:

```
swich (stale) {
    case CDTStaleViewNo:
        return nil;
    case CDTStaleViewOk:
        return [NSURLQueryItem queryItemWithName:@"stale" value:@"ok"];
    case CDTStaleViewUpdateAfter:
        return [NSURLQueryItem queryItemWithName:@"stale" value:@"updateafter"];
}
```

The main reason I like it is because all `switch` cases do the same thing (return a value) rather than some returning and some not, which makes it easier to make mistakes when scan-reading the code.

Either that, or have the first case be `staleString = nil` and guard final return with an `if` statement. Longer, but clearer.